plugins/cros-ec/fu-cros-ec-usb-device.c

```diff
@@ -811,10 +761,13 @@ fu_cros_ec_usb_device_write_firmware(FuDevice *device,
 		}
 
 		/* send start request */
+		fu_cros_ec_first_response_pdu_set_header_type(
+		    st_rpdu,
+		    FU_CROS_EC_FIRST_RESPONSE_PDU_HEADER_TYPE_CR50);
 		if (!fu_device_retry(device,
-				     fu_cros_ec_usb_device_start_request,
-				     SETUP_RETRY_CNT,
-				     &start_resp,
+				     fu_cros_ec_usb_device_start_request_cb,
+				     FU_CROS_EC_SETUP_RETRY_CNT,
+				     &st_rpdu,
```

Comment:
@bleungatchromium this is the bit I wasn't sure about. Because the `START_RESP` struct was init'd with `0x0` before, we get `first_response_pdu.header_type=UPDATE_HEADER_TYPE_CR50` which I've copied here. Is that correct, or should this be `UPDATE_HEADER_TYPE_COMMON` as well? and if `Common`, can we remove the `FuCrosEcFirstResponsePduCr50` struct completely? The data returned as a `FuCrosEcFirstResponsePdu` seems completely unused.

Reply:
It's been a while since I ported this over, but I believe this whole section with UPDATE_HEADER_TYPE_CR50 was taken verbatim from https://chromium.googlesource.com/chromiumos/platform/ec/+/refs/heads/main/include/update_fw.h

Isn't this data structure filled in from the device, though? I'm not sure off the top of my head what the cros_ec usb devices we support actually use (although I would wager common vs cr50).

Reply:
> Isn't this data structure filled in from the device, though

Yes, I think so too. I'll try deleting it and see what happens to the emulation and then real hardware.

Reply:
It seemed to work, lets merge. Thanks for the sanity check.